libraries/core_libs/network/src/tarcap/packets_handlers/votes_sync_packet_handler.cpp

```diff
@@ -54,44 +86,53 @@ void VotesSyncPacketHandler::process(const PacketData &packet_data, const std::s
       disconnect(packet_data.from_node_id_, dev::p2p::UserReason);
       return;
     }
+  }
 
-    if (pbft_current_round < peer_pbft_round) {
-      if (vote_mgr_->voteInVerifiedMap(next_vote)) {
-        LOG(log_dg_) << "Vote " << next_vote_hash.abridged() << " already inserted in verified queue";
-      }
-
-      if (auto vote_is_valid = validateStandardVote(next_vote); vote_is_valid.first == false) {
-        LOG(log_wr_) << "Vote " << next_vote_hash.abridged() << " validation failed. Err: " << vote_is_valid.second;
-        continue;
-      }
-
-      if (!vote_mgr_->addVerifiedVote(next_vote)) {
-        LOG(log_dg_) << "Vote " << next_vote_hash.abridged() << " already inserted in verified queue(race condition)";
-        continue;
-      }
-
-      setVoterMaxPeriodAndRound(vote->getVoterAddr(), vote->getPeriod(), vote->getRound());
-    } else {  // pbft_current_round == peer_pbft_round
-      if (auto vote_is_valid = validateNextSyncVote(next_vote); vote_is_valid.first == false) {
-        LOG(log_wr_) << "Next vote " << next_vote_hash.abridged()
-                     << " validation failed. Err: " << vote_is_valid.second;
-        continue;
-      }
-
-      // CONCERN: Huh? Race condition...
-
-      if (!vote_mgr_->insertUniqueVote(vote)) {
-        LOG(log_dg_) << "Non unique vote " << next_vote_hash.abridged() << " (race condition)";
-        continue;
-      }
-    }
+  // Remove invalid votes from the next_votes bundle. Should we do something with it if
+  auto remove_it = std::remove_if(
```

Comment:
what ? why do we need to remove anything ?

Reply:
previous logic was quite straight forward, in loop either use standard vote validation or next vote validation and then push it into the next votes manager if those were next votes or just regossip if they were standard vote

Reply:
We have checks for whole bundle. And if we see that something wrong with it, then we shouldn't process any votes from it. We can introduce new collection and move processed(valid) elements to it, but I thought that it is cleaner to just remove invalid elements from the first collection that we already have

Reply:
I guess I dont follow